### ISSUE TITLE
ARM Support for promtail and Loki

### ISSUE THREAD

**Issue by Razikus** [grafana/loki#653]
**Is your feature request related to a problem? Please describe.**
Loki and promtail is not available for ARM architecture

**Describe the solution you'd like**
Loki and promtail available on ARM architecture

**Describe alternatives you've considered**
Would be really nice that loki and promtail will have support for ARM devices

**Comment by sh0rez** [comment]
Okay, in theory `promtail` as well as `loki` are already ARM compatible. As they are written in Golang, which is cross platform, they can run everywhere ARM is supported. However, building production binaries is not currently supported using the `Makefile`, as the `assets` task fails because of the wrong `GOARCH`.

Nevertheless you can build development binaries for your platform yourself:

1. Obtain a recent Golang build: https://nats.io/documentation/tutorials/go-install/
2. Checkout loki: `git clone https://github.com/grafana/loki && cd loki`
3. Build promtail: `GOOS=linux GOARCH=arm64 go build ./cmd/promtail`
4. Build loki: `GOOS=linux GOARCH=arm64 go build ./cmd/loki`

Now you have `./loki` and `./promtail` binaries for `arm64`. You may tune `GOOS` for the Operating System (`linux`, `windows`, `darwin`), and `GOARCH` for the architecture (`amd64`, `arm64`, `arm`, etc.)

**Comment by kskewes** [comment]
Have made a start at Docker images for amd64/arm64 images for both [loki](https://gitlab.com/kskewes/loki/blob/master/loki/Dockerfiles/Dockerfile.arm64) and [promtail](https://gitlab.com/kskewes/loki/blob/master/promtail/Dockerfiles/Dockerfile.arm64) as I have a multi-arch k8s cluster.
Still need to work out ports/configuration/etc in Dockerfiles but hopefully this gets you going.

**Comment by RRadziejewski** [comment]
If you have multi arch cluster how do you deploy for example daemon set of promtail ? You are deploying two and in case of arch is amd you use amd image and in case of arm you deploy arm image ?

**Comment by sh0rez** [comment]
Currently, this is probably impossible. However, once we implemented Image Manifest v2, the docker daemon automatically picks the right version for the OS

**Comment by steven-sheehy** [comment]
@Kuqd Duplicate of https://github.com/grafana/loki/issues/67

**Comment by vielmetti** [comment]
@sh0rez what is the obstacle to using Image Manifest v2? Is that an internal dependency, or an external one?

**Comment by sh0rez** [comment]
@vielmetti there is no real obstacle, the required code is just not finished yet. Our current Dockerfile is not cross-platform and the new one is not merged yet (See #668)

Once this is done, I can add manifest v2 support to the CI and cross-platform functionality is provided. Stay tuned!

**Comment by vielmetti** [comment]
Thanks @sh0rez - that clarifies things. (I know that in the past some registries did not support manifest v2, e.g. Quay.io, wanted to distinguish between things within the project and external to it.)

**Comment by sh0rez** [comment]
Hello everyone!
I can proudly announce that this has been finally implemented in b887602e3654d9525467ed0cde79884887d94749 :tada: 

From now on, it is sufficient to just pull the docker image on the desired architecture and it should be the right one.

Currently supported are `amd64`, `arm64` (sometimes called `arm8` or `aarch64`) and `arm7`.
Thanks for your patience!

**Comment by RRadziejewski** [comment]
Gr8 work, keep it up !!

czw., 1.08.2019, 18:19: sh0rez <notifications@github.com> napisał(a):

**Comment by saikrishna1488** [comment]
I faced this issue when i used the ec2 with amazon linux 2 and ARM architecture so follow the below process

wget https://github.com/grafana/loki/releases/download/v2.9.4/loki-linux-arm.zip
unzip loki-linux-arm.zip 
mv loki-linux-arm loki
cp loki /usr/local/bin/
wget https://raw.githubusercontent.com/grafana/loki/main/cmd/loki/loki-local-config.yaml
cp loki-local-config.yaml /usr/local/bin/

vi /etc/systemd/system/loki.service

place the below content in above file 


[Unit]
Description=Loki Service
After=network.target

[Service]
ExecStart=/usr/local/bin/loki -config.file=/usr/local/bin/loki-config.yaml
Restart=always
User=loki
Group=loki

[Install]
WantedBy=multi-user.target

#file end save the file and run below commands

useradd -rs /bin/false loki
systemctl daemon-reload
sudo systemctl enable loki
sudo systemctl start loki
sudo systemctl status loki
